pymc3/distributions/timeseries.py

```diff
@@ -464,6 +464,86 @@ def logp(self, x):
     def _distr_parameters_for_repr(self):
         return ["mu", "cov"]
 
+    def random(self, point=None, size=None):
+        """
+        Draw random values from MvGaussianRandomWalk.
+
+        Parameters
+        ----------
+        point: dict, optional
+            Dict of variable values on which random values are to be
+            conditioned (uses default point if not specified).
+        size: int, optional
+            Desired size of random sample (returns one sample if not
+            specified).
+
+        Returns
+        -------
+        array
+
+
+        Examples
+        -------
+        Create one sample from a 2-dimensional Gaussian random walk with 10 timesteps::
+
+            mu = np.array([1.0, 0.0])
+            cov = np.array([[1.0, 0.0], [0.0, 2.0]])
+            sample = MvGaussianRandomWalk(mu, cov, shape=(10, 2)).random(size=1)
+        """
+
+        param_attribute = getattr(
+            self.innov, "chol_cov" if self.innov._cov_type == "chol" else self.innov._cov_type
+        )
+        mu, param = distribution.draw_values(
+            [self.innov.mu, param_attribute], point=point, size=size
+        )
+        return distribution.generate_samples(
+            self._random,
+            size=size,
+            dist_shape=self.shape,
+            not_broadcast_kwargs={
+                "sample_shape": to_tuple(size),
+                "param": param,
+                "mu": mu,
+                "cov_type": self.innov._cov_type,
+            },
+        )
+
+    def _random(self, mu, param, size, sample_shape, cov_type):
+        """
+        Implements the multivariate Gaussian random walk as a cumulative
+        sum of i.i.d. multivariate Gaussians.
+        Assumes that
+        size is of the form (samples, time, dims).
+        """
+
+        if cov_type == "chol":
+            cov = np.matmul(param, param.transpose())
+        elif cov_type == "tau":
+            cov = np.linalg.inv(param)
+        else:
+            cov = param
+
+        # time axis comes after the sample axis
+        time_axis = len(sample_shape)
+
```

Comment:
What if sample shape is None? Then time_axis = 0. 
Is this correct behaviour?

Reply:
Yes, that was my intent in my version of `_random`. My understanding was that the `_random` function will receive sample size of None when the user calls `MvGaussianRandomWalk.random()` without specifying size. In this case I think the shape of the random walk will be (timesteps, spacial_dims), so having time_axis = 0 would be right.

Reply:
Makes sense 👍